### PR TITLE
WT-9089 Define a new remove operation function in the cpp testing

### PR DESCRIPTION
**Pull request by etienneptl**
A new `remove_operation` function has been added to the `databaes_operation` class. This means one can override the behavior of this function at the test level and parameterize it through the test configuration file.

### CODE REVIEW

File: dist/test_data.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def __ge__(self, other):
     Config('ops_per_transaction', '', r'''
         Defines how many operations a transaction can perform, the range is defined with a minimum
         and a maximum and a random number is chosen between the two using a linear distribution.''',
-        type='category',subconfig=range_config),
+        type='category', subconfig=range_config),
```
Comment (luke-pearson) [inline dist/test_data.py:113]: nice 👀

File: test/cppsuite/configs/operations_test_stress.txt [hunk 2]
```diff
@@ -50,6 +50,12 @@ workload_generator=
         ops_per_transaction=(max=100,min=50),
         thread_count=20
     ),
+    remove_config=
+    (
+        op_rate=100ms,
+        ops_per_transaction=(max=20,min=10),
```
Comment (luke-pearson) [inline test/cppsuite/configs/operations_test_stress.txt:56]: Lets make min, max, 50, 100 so its a bit more stressful.

File: test/cppsuite/test_harness/workload/database_operation.cpp [hunk 3]
```diff
@@ -246,6 +246,75 @@ database_operation::read_operation(thread_context *tc)
...
+        tc->transaction.try_begin();
+
+        /* Get the cursor associated with the collection. */
+        scoped_cursor &rnd_cursor = rnd_cursors[coll.id];
+        scoped_cursor &cursor = cursors[coll.id];
+
+        /* Choose a random key to delete. */
+        const char *key_str;
+        int ret = rnd_cursor->next(rnd_cursor.get());
+        /* It is possible not to find anything if the collection is empty. */
```
Comment (luke-pearson) [inline test/cppsuite/test_harness/workload/database_operation.cpp:296]: Is this going to blow up our update_operation? I.e. if we try and update a removed record, or does WiredTiger allow updates to become inserts?

Comment (etienneptl) [reply test/cppsuite/test_harness/workload/database_operation.cpp:296]: Before calling `update` on a key, we could check if it exists to make sure we are updating it. If it does not exist, we could call `insert` in the `update_operation` or just get another key.
```
        /* Choose a random key to update. */
        uint64_t key_id =
          random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
        if(key_id.exists())
            if (!tc->update(cursor, coll.id, tc->key_to_string(key_id))) {
                tc->transaction.rollback();
            }
       else
            if (!tc->insert(cursor, coll.id, tc->key_to_string(key_id))) {
                tc->transaction.rollback();
            }
```

File: test/cppsuite/test_harness/workload/database_operation.cpp [hunk 4]
```diff
@@ -246,6 +246,75 @@ database_operation::read_operation(thread_context *tc)
...
+        testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+        if (ret == WT_NOTFOUND)
+            continue;
+        testutil_check(rnd_cursor->get_key(rnd_cursor.get(), &key_str));
+        if (!tc->remove(cursor, coll.id, key_str)) {
```
Comment (jiechenbo) [inline test/cppsuite/test_harness/workload/database_operation.cpp:301]: Can we just re-use the rnd_cursor here to perform remove?

File: test/cppsuite/configs/operations_test_default.txt [hunk 5]
```diff
@@ -65,6 +65,12 @@ workload_generator=
         ops_per_transaction=(max=100,min=50),
         thread_count=10
     ),
+    remove_config=
+    (
+        op_rate=500ms,
+        ops_per_transaction=(max=100,min=0),
```
Comment (jiechenbo) [inline test/cppsuite/configs/operations_test_default.txt:71]: We should make max less, since its more of a default config.

**Review by jiechenbo — COMMENTED** [review]
Just some np questions! Generally nice work!! I am curious if the tracking operation for REMOVE looks good with this change? Just confirming!

File: test/cppsuite/test_harness/workload/database_operation.cpp (hunk 4 above)
Comment (etienneptl) [reply test/cppsuite/test_harness/workload/database_operation.cpp:301]: We cannot as this operation is not supported by the cursor, I added a comment [here](https://github.com/wiredtiger/wiredtiger/pull/7822/files#diff-d35d5f5a8892ee5cd7efdd7079e296da18b9223d0b563b5f92cc8d65f8fa8acaR255-R259).

**Review by jiechenbo — APPROVED** [review]
LGTM!

**Comment by etienneptl** [comment]
Sorry, last commit is a mistake.

File: test/cppsuite/configs/operations_test_stress.txt (hunk 2 above)
Comment (etienneptl) [reply test/cppsuite/configs/operations_test_stress.txt]: Done in 2b4671a321ed4c502730d86ee40d72e849388761

File: test/cppsuite/configs/operations_test_default.txt (hunk 5 above)
Comment (etienneptl) [reply test/cppsuite/configs/operations_test_default.txt]: Done in b8066f503c28d155e09f187b88472a268d760ca5

File: test/cppsuite/test_harness/workload/database_operation.cpp (hunk 3 above)
Comment (etienneptl) [reply test/cppsuite/test_harness/workload/database_operation.cpp:296]: Yes, WiredTiger allows updates to become inserts. If the record does not exist and we call `update`, it becomes an insert. Hence the suggestion I made just above is not worth it.

I had to finish the transaction of the remove thread if no records is found so we can get a chance to find newer ones, see bef13fdfb943f0ce6888dd6f59a5fbf3eccffb1a.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**